### PR TITLE
fix SerialPort

### PR DESCRIPTION
**Pull request by vwaurich**
fixes and enhancement for the win implementation of serialPort
fix: 
- put whole buffer in MDD_SerialPackagerSetDataWithErrorReturn()
- append the messages to the buffer until its full, dont start from idx0 with every ReadFile() since this messes up messages which come in fractions

enhancement:
- copy complete messages to an extra buffer and put this one in the packager

I just tested this with a SerialReceive and and some unpackInt blocks. If you have any further tests, please check if everything is working.

### CODE REVIEW

File: Modelica_DeviceDrivers/Resources/Include/MDDSerialPort.h [hunk 1]
```diff
@@ -101,7 +102,13 @@ DWORD WINAPI MDD_serialPortReceivingThread(LPVOID p_serial) {
                         if (x > serial->bufferSize) {
                             x = (DWORD)serial->bufferSize;
                         }
-                        ret = ReadFile(serial->hComm, serial->receiveBuffer, x, &serial->receivedBytes, &rdSync);
+                        ret = ReadFile(serial->hComm, &serial->receiveBuffer[idx], x, &serial->receivedBytes, &rdSync);											
+                        idx = idx + serial->receivedBytes;
+                        if(idx == serial->bufferSize)
+                        {
+                          memcpy(serial->completeMessage, serial->receiveBuffer, serial->bufferSize);
+                          idx = 0;
```
Comment (tbeu) [inline Modelica_DeviceDrivers/Resources/Include/MDDSerialPort.h]: Why do you copy bufferSize bytes and not just append the receivedBytes?

**Comment by tbeu** [comment]
Why do you copy bufferSize bytes and not just append the receivedBytes?

File: Modelica_DeviceDrivers/Resources/Include/MDDSerialPort.h (hunk 1 above)
Comment (vwaurich) [inline Modelica_DeviceDrivers/Resources/Include/MDDSerialPort.h]: The message comes in fractions and is stored byte per byte in the receiveBuffer. If bufferSzze bytes have been read, the message is complete used for the packager functions to unpack it. The message fractions don't necessarily come in accordance with the datatypes I want to unpack. If MDD_serialPortReadP is called receiveBuffer might be a mixture of the last and the current message. completeMessage always contains an associated message.

Comment (tbeu) [inline Modelica_DeviceDrivers/Resources/Include/MDDSerialPort.h]: Ah, I see. You only copy once it is completely filled.

Will merge it soon. Thanks.

**Comment by tbeu** [comment]
Thank you for the PR. There was some potential error in the available size `x` of the receive buffer. I fixed this problem and also the other problems mentioned in #127 by #133 (thus superseding this PR #132) and run the test model TestSerialPackager_SerialPort with fixed buffer size of 64 succesfully on Windows using the com0com emulator.

**Comment by bernhard-thiele** [comment]
@vwaurich Since this PR is apparently superseded by the one above, I will close the ticket (please complain if it should stay open)

**Comment by vwaurich** [comment]
Alright. No problem.

**Pull request closed without merging**